quipucords/scanner/network/inspect.py

```diff
@@ -79,7 +84,24 @@ def run(self):
 
         try:
             # Execute scan
-            self.inspect_scan()
+            connected, failed, completed = self.obtain_discovery_data()
+            num_completed = len(completed)
+            num_remaining = len(connected)
+            num_total = num_remaining + num_completed
+            num_failed = len(failed)
+
+            if num_total == 0:
+                msg = 'Inventory provided no reachable hosts.'
+                raise ScannerException(msg)
+
+            logger.info('Inspect scan task started for %s.',
+                        self.scan_task)
+            log_msg = '%d total connected, %d completed, %d'\
+                ' remaining, and make %d failed hosts'
+            logger.info(log_msg,
+                        num_total, num_completed, num_remaining, num_failed)
+
+            self.inspect_scan(connected)
```

Comment:
What was the point in moving this block of code out of `inspect_scan` and changing the `inspect_scan` signature?  Testability?

Reply:
Yes, testability. Otherwise the test would have been much more convoluted.